src/Form.js

```diff
@@ -12,6 +13,17 @@ class Form extends Component {
     this.context._reduxForm.registerInnerOnSubmit(this.props.onSubmit)
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.onChange) {
+      const currentValues = this.context._reduxForm.getValues()
+      if (this.previousValues) {
+        const currentValuesDiffer = !deepEqual(currentValues, this.previousValues)
+        if (currentValuesDiffer) nextProps.onChange(currentValues)
```

Comment:
I would call `onChange` even for the first call (when `this.previousValues` is falsy)